### PR TITLE
Fix mount directory for HDDL sockets and forward text output from openvino sample application

### PR DESCRIPTION
**Pull request by alvarovn**
This PR aims to fix a few issues that I've encountered related to the Openvino Sample Application.

The kubernetes manifest for the consumer app mounts `/tmp` directory. I'm assuming that the reason to mount this directory is to get access to the `hddl-daemon` sockets. However, that directory is actually `/var/tmp`. See:
https://docs.openvinotoolkit.org/latest/openvino_docs_install_guides_installing_openvino_docker_linux.html
https://github.com/open-ness/specs/blob/master/doc/building-blocks/enhanced-platform-awareness/openness_hddl.md
https://github.com/open-ness/openness-experience-kits/blob/master/roles/hddl/network_edge/controlplane/templates/setup_hddl_daemonset.yaml.j2#L41
Also the `/dev/shm` mount is missing, which is another requirement to work with HDDL daemon.

Moreover, every time that the application attempted to connect to HDDL daemon,  it made the daemon crash with a segmentation fault. Following this thread: https://community.intel.com/t5/Intel-Distribution-of-OpenVINO/Segmentation-fault-for-object-detection-sample-with-MYRIAD/td-p/1182239
I changed the patch file and the issue disappeared.

Furthermore, I was getting the following issue:
```
[ ERROR ] Failed to create plugin /opt/intel/openvino_2020.2.120/deployment_tools/inference_engine/lib/intel64/libHDDLPlugin.so for device HDDL
Please, check your environment
Cannot load library '/opt/intel/openvino_2020.2.120/deployment_tools/inference_engine/lib/intel64/libHDDLPlugin.so': libjson-c.so.2: cannot open shared object file: No such file or directory
```
Which is fixed by adding the missing library as a dependency in the dockerfile.

Finally, the stdout and stderr from the C++ openvino application is not being shown correctly in the Go application. Since the main function calling the go routines that control the text forwarding do not wait for them to finish, nothing gets forwarded. I added a waitgroup to control that.
This last fix was needed to be able to see if the C++ openvino app is detecting the HDDL daemon or not.

Let me know if this should be divided into multiple PRs.

### CODE REVIEW

**Review by jiangzhg — APPROVED** [review]

File: applications/openvino/consumer/cmd/openvino.go [hunk 1]
```diff
@@ -59,21 +60,32 @@ func callOpenVINO(model string, accl string) {
 		"-i", "rtp://127.0.0.1:5000?overrun_nonfatal=1",
 		"-m", modelXML)
```
Comment (jiangzhg) [inline applications/openvino/consumer/cmd/openvino.go:62]: there is an extra 'tab' blank.

File: applications/openvino/consumer/cmd/openvino.go [hunk 2]
```diff
@@ -59,21 +60,32 @@ func callOpenVINO(model string, accl string) {
...
 
+	stdout, _ := cmd.StdoutPipe()
+	stderr, _ := cmd.StderrPipe()
+
```
Comment (jiangzhg) [inline applications/openvino/consumer/cmd/openvino.go:65]: there is an extra 'tab' blank.

File: applications/openvino/consumer/cmd/openvino.go [hunk 3]
```diff
@@ -59,21 +60,32 @@ func callOpenVINO(model string, accl string) {
...
 		}
+		wg.Done()
 	}()
 
-	err := cmd.Start()
-	if err != nil {
+	if err := cmd.Start(); err != nil {
 		log.Fatal("Failed to run OpenVINO process:", err)
 	}
+
```
Comment (jiangzhg) [inline applications/openvino/consumer/cmd/openvino.go:85]: there is an extra 'tab' blank, you can run this command , "gofmt -d cmd/openvino.go", to check.

**Comment by jiangzhg** [comment]
diff --git a/applications/openvino/benchmark/do_benchmark.sh b/applications/openvino/benchmark/do_benchmark.sh
index 4deba7d..c754f36 100755
--- a/applications/openvino/benchmark/do_benchmark.sh
+++ b/applications/openvino/benchmark/do_benchmark.sh
@@ -3,8 +3,9 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright (c) 2020 Intel Corporation

+# shellcheck disable=SC1091
 source /opt/intel/openvino/bin/setupvars.sh
-cd /opt/intel/openvino/deployment_tools/demo
+cd /opt/intel/openvino/deployment_tools/demo || exit
 ./demo_squeezenet_download_convert_run.sh -d "${TARGET_DEVICE}"

 /root/inference_engine_samples_build/intel64/Release/benchmark_app -i "${IMAGE}" -m "${MODEL}" -d "${TARGET_DEVICE}" -nireq "${NIREQ}" -niter "${NITER}" -api "${API}" -b "${BATCH_SIZE}"
diff --git a/applications/openvino/consumer/cmd/openvino.go b/applications/openvino/consumer/cmd/openvino.go
index a77226c..d362dc2 100644
--- a/applications/openvino/consumer/cmd/openvino.go
+++ b/applications/openvino/consumer/cmd/openvino.go
@@ -62,7 +62,7 @@ func callOpenVINO(model string, accl string) {

        stdout, _ := cmd.StdoutPipe()
        stderr, _ := cmd.StderrPipe()
-
+
        var wg sync.WaitGroup
        wg.Add(2)

@@ -82,7 +82,7 @@ func callOpenVINO(model string, accl string) {
        if err := cmd.Start(); err != nil {
                log.Fatal("Failed to run OpenVINO process:", err)
        }
-
+
        wg.Wait()

        if err := cmd.Wait(); err != nil {
diff --git a/applications/openvino/consumer/start.sh b/applications/openvino/consumer/start.sh
index 48356eb..3e15836 100755
--- a/applications/openvino/consumer/start.sh
+++ b/applications/openvino/consumer/start.sh
@@ -3,6 +3,7 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright (c) 2019 Intel Corporation

+# shellcheck disable=SC1091
 source /opt/intel/openvino/bin/setupvars.sh
 echo "192.168.1.10 analytics.openness" >> /etc/hosts
 ./fwd.sh &
diff --git a/applications/openvino/producer/cmd/eaa_interface.go b/applications/openvino/producer/cmd/eaa_interface.go
index dc75a6b..6f42911 100644
--- a/applications/openvino/producer/cmd/eaa_interface.go
+++ b/applications/openvino/producer/cmd/eaa_interface.go
@@ -55,12 +55,12 @@ type URN struct {
 // is used, with any newlines indicated with `\n` characters. Most languages
 // provide encoders that correctly marshal this out. For more information,
 // see the RFC here: https://tools.ietf.org/html/rfc7468
-type AuthCredentials struct {
-       ID          string   `json:"id,omitempty"`
-       Certificate string   `json:"certificate,omitempty"`
-       CaChain     []string `json:"ca_chain,omitempty"`
-       CaPool      []string `json:"ca_pool,omitempty"`
-}
+//type AuthCredentials struct {
+//     ID          string   `json:"id,omitempty"`
+//     Certificate string   `json:"certificate,omitempty"`
+//     CaChain     []string `json:"ca_chain,omitempty"`
+//     CaPool      []string `json:"ca_pool,omitempty"`
+//}

 // AuthIdentity defines a request to obtain authentication credentials. These
 // credentials would be used to further communicate with endpoint(s) that are
@@ -70,6 +70,6 @@ type AuthCredentials struct {
 // is used, with any newlines indicated with `\n` characters. Most languages
 // provide encoders that correctly marshal this out. For more information,
 // see the RFC here: https://tools.ietf.org/html/rfc7468
-type AuthIdentity struct {
-       Csr string `json:"csr,omitempty"`
-}
+//type AuthIdentity struct {
+///    Csr string `json:"csr,omitempty"`
+//}
diff --git a/applications/openvino/producer/cmd/main.go b/applications/openvino/producer/cmd/main.go
index a6a4acc..174c63d 100644
--- a/applications/openvino/producer/cmd/main.go
+++ b/applications/openvino/producer/cmd/main.go
@@ -67,6 +67,7 @@ func createEncryptedClient() (*http.Client, error) {
                                RootCAs:      certPool,
                                Certificates: []tls.Certificate{cert},
                                ServerName:   EAACommonName,
+                               MinVersion:   tls.VersionTLS12,
                        },
                },
                Timeout: 0,
(END)

**Comment by jiangzhg** [comment]
@alvarovn  Please refer to the following changes and submit again.
`From 6fc9270579f14bdc0ffec07f8aebe55571fa6a2c Mon Sep 17 00:00:00 2001
From: Jiang Zhenggang <zhenggang1x.jiang@intel.com>
Date: Tue, 2 Feb 2021 16:21:52 +0800
Subject: [PATCH] fix some check issues by CI tools

---
 .../openvino/benchmark/do_benchmark.sh         |  3 ++-
 applications/openvino/consumer/cmd/openvino.go |  4 ++--
 applications/openvino/consumer/start.sh        |  1 +
 .../openvino/producer/cmd/eaa_interface.go     | 18 +++++++++---------
 applications/openvino/producer/cmd/main.go     |  1 +
 5 files changed, 15 insertions(+), 12 deletions(-)

diff --git a/applications/openvino/benchmark/do_benchmark.sh b/applications/openvino/benchmark/do_benchmark.sh
index 4deba7d..c754f36 100755
--- a/applications/openvino/benchmark/do_benchmark.sh
+++ b/applications/openvino/benchmark/do_benchmark.sh
@@ -3,8 +3,9 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright (c) 2020 Intel Corporation

+# shellcheck disable=SC1091
 source /opt/intel/openvino/bin/setupvars.sh
-cd /opt/intel/openvino/deployment_tools/demo
+cd /opt/intel/openvino/deployment_tools/demo || exit
 ./demo_squeezenet_download_convert_run.sh -d "${TARGET_DEVICE}"

 /root/inference_engine_samples_build/intel64/Release/benchmark_app -i "${IMAGE}" -m "${MODEL}" -d "${TARGET_DEVICE}" -nireq "${NIREQ}" -niter "${NITER}" -api "${API}" -b "${BATCH_SIZE}"
diff --git a/applications/openvino/consumer/cmd/openvino.go b/applications/openvino/consumer/cmd/openvino.go
index a77226c..d362dc2 100644
--- a/applications/openvino/consumer/cmd/openvino.go
+++ b/applications/openvino/consumer/cmd/openvino.go
@@ -62,7 +62,7 @@ func callOpenVINO(model string, accl string) {

        stdout, _ := cmd.StdoutPipe()
        stderr, _ := cmd.StderrPipe()
-
+
        var wg sync.WaitGroup
        wg.Add(2)

@@ -82,7 +82,7 @@ func callOpenVINO(model string, accl string) {
        if err := cmd.Start(); err != nil {
                log.Fatal("Failed to run OpenVINO process:", err)
        }
-
+
        wg.Wait()

        if err := cmd.Wait(); err != nil {
diff --git a/applications/openvino/consumer/start.sh b/applications/openvino/consumer/start.sh
index 48356eb..3e15836 100755
--- a/applications/openvino/consumer/start.sh
+++ b/applications/openvino/consumer/start.sh
@@ -3,6 +3,7 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright (c) 2019 Intel Corporation

+# shellcheck disable=SC1091
 source /opt/intel/openvino/bin/setupvars.sh
 echo "192.168.1.10 analytics.openness" >> /etc/hosts
 ./fwd.sh &
diff --git a/applications/openvino/producer/cmd/eaa_interface.go b/applications/openvino/producer/cmd/eaa_interface.go
index dc75a6b..6f42911 100644
--- a/applications/openvino/producer/cmd/eaa_interface.go
+++ b/applications/openvino/producer/cmd/eaa_interface.go
@@ -55,12 +55,12 @@ type URN struct {
 // is used, with any newlines indicated with `\n` characters. Most languages
 // provide encoders that correctly marshal this out. For more information,
 // see the RFC here: https://tools.ietf.org/html/rfc7468
-type AuthCredentials struct {
-       ID          string   `json:"id,omitempty"`
-       Certificate string   `json:"certificate,omitempty"`
-       CaChain     []string `json:"ca_chain,omitempty"`
-       CaPool      []string `json:"ca_pool,omitempty"`
-}
+//type AuthCredentials struct {
+//     ID          string   `json:"id,omitempty"`
+//     Certificate string   `json:"certificate,omitempty"`
+//     CaChain     []string `json:"ca_chain,omitempty"`
+//     CaPool      []string `json:"ca_pool,omitempty"`
+//}

 // AuthIdentity defines a request to obtain authentication credentials. These
 // credentials would be used to further communicate with endpoint(s) that are
@@ -70,6 +70,6 @@ type AuthCredentials struct {
 // is used, with any newlines indicated with `\n` characters. Most languages
 // provide encoders that correctly marshal this out. For more information,
 // see the RFC here: https://tools.ietf.org/html/rfc7468
-type AuthIdentity struct {
-       Csr string `json:"csr,omitempty"`
-}
+//type AuthIdentity struct {
+///    Csr string `json:"csr,omitempty"`
+//}
diff --git a/applications/openvino/producer/cmd/main.go b/applications/openvino/producer/cmd/main.go
index a6a4acc..174c63d 100644
--- a/applications/openvino/producer/cmd/main.go
+++ b/applications/openvino/producer/cmd/main.go
@@ -67,6 +67,7 @@ func createEncryptedClient() (*http.Client, error) {
                                RootCAs:      certPool,
                                Certificates: []tls.Certificate{cert},
                                ServerName:   EAACommonName,
+                               MinVersion:   tls.VersionTLS12,
                        },
                },
                Timeout: 0,
--
2.17.1
`

**Comment by jiangzhg** [comment]
@alvarovn  Please refer to the following changes and submit again

`From 6fc9270579f14bdc0ffec07f8aebe55571fa6a2c Mon Sep 17 00:00:00 2001
From: Jiang Zhenggang <zhenggang1x.jiang@intel.com>
Date: Tue, 2 Feb 2021 16:21:52 +0800
Subject: [PATCH] fix some check issues by CI tools

---
 .../openvino/benchmark/do_benchmark.sh         |  3 ++-
 applications/openvino/consumer/cmd/openvino.go |  4 ++--
 applications/openvino/consumer/start.sh        |  1 +
 .../openvino/producer/cmd/eaa_interface.go     | 18 +++++++++---------
 applications/openvino/producer/cmd/main.go     |  1 +
 5 files changed, 15 insertions(+), 12 deletions(-)

diff --git a/applications/openvino/benchmark/do_benchmark.sh b/applications/openvino/benchmark/do_benchmark.sh
index 4deba7d..c754f36 100755
--- a/applications/openvino/benchmark/do_benchmark.sh
+++ b/applications/openvino/benchmark/do_benchmark.sh
@@ -3,8 +3,9 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright (c) 2020 Intel Corporation

+# shellcheck disable=SC1091
 source /opt/intel/openvino/bin/setupvars.sh
-cd /opt/intel/openvino/deployment_tools/demo
+cd /opt/intel/openvino/deployment_tools/demo || exit
 ./demo_squeezenet_download_convert_run.sh -d "${TARGET_DEVICE}"

 /root/inference_engine_samples_build/intel64/Release/benchmark_app -i "${IMAGE}" -m "${MODEL}" -d "${TARGET_DEVICE}" -nireq "${NIREQ}" -niter "${NITER}" -api "${API}" -b "${BATCH_SIZE}"
diff --git a/applications/openvino/consumer/cmd/openvino.go b/applications/openvino/consumer/cmd/openvino.go
index a77226c..d362dc2 100644
--- a/applications/openvino/consumer/cmd/openvino.go
+++ b/applications/openvino/consumer/cmd/openvino.go
@@ -62,7 +62,7 @@ func callOpenVINO(model string, accl string) {

        stdout, _ := cmd.StdoutPipe()
        stderr, _ := cmd.StderrPipe()
-
+
        var wg sync.WaitGroup
        wg.Add(2)

@@ -82,7 +82,7 @@ func callOpenVINO(model string, accl string) {
        if err := cmd.Start(); err != nil {
                log.Fatal("Failed to run OpenVINO process:", err)
        }
-
+
        wg.Wait()

        if err := cmd.Wait(); err != nil {
diff --git a/applications/openvino/consumer/start.sh b/applications/openvino/consumer/start.sh
index 48356eb..3e15836 100755
--- a/applications/openvino/consumer/start.sh
+++ b/applications/openvino/consumer/start.sh
@@ -3,6 +3,7 @@
 # SPDX-License-Identifier: Apache-2.0
 # Copyright (c) 2019 Intel Corporation

+# shellcheck disable=SC1091
 source /opt/intel/openvino/bin/setupvars.sh
 echo "192.168.1.10 analytics.openness" >> /etc/hosts
 ./fwd.sh &
diff --git a/applications/openvino/producer/cmd/eaa_interface.go b/applications/openvino/producer/cmd/eaa_interface.go
index dc75a6b..6f42911 100644
--- a/applications/openvino/producer/cmd/eaa_interface.go
+++ b/applications/openvino/producer/cmd/eaa_interface.go
@@ -55,12 +55,12 @@ type URN struct {
 // is used, with any newlines indicated with `\n` characters. Most languages
 // provide encoders that correctly marshal this out. For more information,
 // see the RFC here: https://tools.ietf.org/html/rfc7468
-type AuthCredentials struct {
-       ID          string   `json:"id,omitempty"`
-       Certificate string   `json:"certificate,omitempty"`
-       CaChain     []string `json:"ca_chain,omitempty"`
-       CaPool      []string `json:"ca_pool,omitempty"`
-}
+//type AuthCredentials struct {
+//     ID          string   `json:"id,omitempty"`
+//     Certificate string   `json:"certificate,omitempty"`
+//     CaChain     []string `json:"ca_chain,omitempty"`
+//     CaPool      []string `json:"ca_pool,omitempty"`
+//}

 // AuthIdentity defines a request to obtain authentication credentials. These
 // credentials would be used to further communicate with endpoint(s) that are
@@ -70,6 +70,6 @@ type AuthCredentials struct {
 // is used, with any newlines indicated with `\n` characters. Most languages
 // provide encoders that correctly marshal this out. For more information,
 // see the RFC here: https://tools.ietf.org/html/rfc7468
-type AuthIdentity struct {
-       Csr string `json:"csr,omitempty"`
-}
+//type AuthIdentity struct {
+///    Csr string `json:"csr,omitempty"`
+//}
diff --git a/applications/openvino/producer/cmd/main.go b/applications/openvino/producer/cmd/main.go
index a6a4acc..174c63d 100644
--- a/applications/openvino/producer/cmd/main.go
+++ b/applications/openvino/producer/cmd/main.go
@@ -67,6 +67,7 @@ func createEncryptedClient() (*http.Client, error) {
                                RootCAs:      certPool,
                                Certificates: []tls.Certificate{cert},
                                ServerName:   EAACommonName,
+                               MinVersion:   tls.VersionTLS12,
                        },
                },
                Timeout: 0,
--
2.17.1`

**Comment by jiangzhg** [comment]
@alvarovn  Please refer to the following changes and submit again.

[https://raw.githubusercontent.com/jiangzhg/tmppatch/master/0001-fix-some-check-issues-by-CI-tools.patch](url)

File: applications/openvino/consumer/object_detection_demo_ssd_async.patch [hunk 4]
```diff
@@ -1,6 +1,6 @@
---- main.cpp	2019-09-19 18:11:55.607355860 +0100
-+++ main.cpp_changed	2019-09-19 18:11:46.981355924 +0100
-@@ -79,6 +79,10 @@
+--- mainnewer.cpp	2021-01-28 17:28:51.064453475 +0100
++++ "mainnewer copy.cpp"	2021-01-28 17:37:07.138143939 +0100
```
Comment (alvarovn) [inline applications/openvino/consumer/object_detection_demo_ssd_async.patch]: These file names changed and they shouldn't. I'll apply this change too.

**Review by abjyoti — APPROVED** [review]
I am ok..

**Review by jiangzhg — APPROVED** [review]

**Review by tongtongxiaopeng1979 — APPROVED** [review]

**Pull request merged**